zombienet_tests/functional/0003-beefy-finalized-heads.js

```diff
@@ -0,0 +1,23 @@
+async function run(_, networkInfo, nodeNames) {
+  const apis = await Promise.all(
+    nodeNames.map(async (nodeName) => {
+      const { wsUri, userDefinedTypes } = networkInfo.nodesByName[nodeName];
+      return await zombie.connect(wsUri, userDefinedTypes);
+    })
+  );
+
+  const finalizedHeads = await Promise.all(
+    apis.map(async (api) => {
+      return api.rpc.beefy.getFinalizedHead();
+    })
+  );
+
+  // check that the finalized heads are the same
+  if (finalizedHeads.every((finalizedHead) => finalizedHead.toString() === finalizedHeads[0].toString())) {
```

Comment:
> I suggest instead to check that RPC provided finalizedHead hash is for block number > 21 (or some number we know should be finalized by now)

I've watered the strict equality down to checking whether the finalized head's heights have distance of at most 1 from the first validator's: https://github.com/paritytech/polkadot/pull/7068/commits/c01e3f38022d9c4a827e0d3b0cb1c93e038c2b8e

Reply:
[minimum BEEFY finality block delta](https://github.com/paritytech/polkadot/blob/67278e018236dad1aa42d512791e114479fbc043/node/service/src/lib.rs#L1154) is `8` for `Polkadot` network on which these zombienet tests run AFAIK. So the distance will never be `1` (except special case where some N is finalized and N+1 turns out to be mandatory block).

But in your new version of the code, you are getting block number so I suggest just checking it's `> 21` (since we already explicitly check best finalized is > 21 in node metrics, it makes sense that RPC reported `finalizedHead` should also report something > 21).

Reply:
good catch re. block delta's; changed to just checking block height `≥ 21` here: https://github.com/paritytech/polkadot/pull/7068/commits/89ced7dd934d66a6eebdfab3abcff3c47db010ac

I was somewhat dissatisfied with essentially reproducing what the `substrate_beefy_best_block` test already covers (outside of checking the API call itself), so https://github.com/paritytech/polkadot/pull/7068/commits/324cb73e77b0bef7b728a9212aa1fa2c106a7967 gets back what I originally intended (ensuring that the finalized heads reported by different nodes all share a canonical chain) without being fallible.